src/legacy/resolver.rs

```diff
@@ -122,5 +122,8 @@ pub fn typedef<TArg: Type, TResult: Type>(defs: &mut TypeMap) -> ProcedureDataTy
     let mut type_map = TypeMap::default();
     let arg_ty = TArg::reference(&mut type_map, &[]).inner;
     let result_ty = TResult::reference(&mut type_map, &[]).inner;
+
+    defs.append(&mut type_map);
```

Comment:
We would be better off removing the `type_map` variable and then using `defs` instead of it.

Would achieve the same effect but cleaner and probally slightly more efficiently.

Reply:
Ahh, somehow missed that, thanks!